### PR TITLE
File io

### PR DESCRIPTION
**Pull request by Alan-Choi**
Let me know if there are any issues.

### CODE REVIEW

File: src/main/java/LocalStorageTranslator.java [hunk 1]
```diff
@@ -171,4 +173,16 @@ public void store(ArrayList<String> _params, ArrayList<String> _input) {
...
+
+    private static String getFileName(ArrayList<String> _storageInfo) {
+        if(_storageInfo.isEmpty()) {
+            throw new IllegalArgumentException("Empty arraylist!");
+        } else {
+            return _storageInfo.get(0);
+        }
+    }
+
+    private static Path getPath(String _fileName) {
```
Comment (anthonymesa) [inline src/main/java/LocalStorageTranslator.java:185]: lets implement a try catch here for InvalidPathException

File: src/main/java/DataStorage.java [hunk 2]
```diff
@@ -30,8 +34,13 @@ public DataStorage(StorageType _type) {
      * @param _input  ArrayList of Entries to be stored to desired location.
      */
     public void save(ArrayList<String> _params, ArrayList<Entry> _input) {
```
Comment (anthonymesa) [inline src/main/java/DataStorage.java:36]: Does this compile? Shouldn't there be a "throws" declaration in this function definition?

File: src/main/java/LocalStorageTranslator.java [hunk 3]
```diff
@@ -98,30 +98,30 @@ public void saveSettings(Map<String, String> _settings, ArrayList<String> _param
...
+                    writer.println(_input.get(i));
                 }
             }
-//            pw.println(_input.toString());
-            pw.flush();
-            pw.close();
+
+            writer.flush();
+            writer.close();
         } catch (IOException ex) {
```
Comment (anthonymesa) [inline src/main/java/LocalStorageTranslator.java:125]: If storing the file fails, lets throw an exception.

File: src/main/java/LocalStorageTranslator.java [hunk 4]
```diff
@@ -137,27 +137,29 @@ public void store(ArrayList<String> _params, ArrayList<String> _input) {
...
                 double low = read.nextDouble();
                 double high = read.nextDouble();
                 int volume = read.nextInt();
-                date = format.parse(read.nextLine());
+                Date date = format.parse(read.nextLine());
 
+                // The usable data is retrieved as instances of Entries and are stored in the ArrayList of Objects
                 Entry entry = new Entry(open, close, low, high, volume, date);
                 data.add(entry);
             }
```
Comment (anthonymesa) [inline src/main/java/LocalStorageTranslator.java:165]: again, lets catch any errors and throw an error

**Review by anthonymesa — CHANGES_REQUESTED** [review]
Great work! I left some comments about implementing more try/catch's. The functions that throw exceptions will need to have "throws ExceptionOfSomeType" in their function declarations, and these function declarations will also need to be updated to match in the StorageAdapter.

File: src/main/java/DataStorage.java (hunk 2 above)
Comment (Alan-Choi) [reply src/main/java/DataStorage.java:36]: It does compile and should run just fine, but I can put the declaration if you'd like.

**Review by anthonymesa — APPROVED** [review]

**Pull request merged**